src/libs/E2E/API.mock.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable rulesdir/no-api-in-views */
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import Log from '../Log';
+
+/**
+ * A dictionary which has the name of a API command as key, and a function which
+ * receives the api command parameters as value and is expected to return a response
+ * object.
+ */
+const mocks = {
+    BeginSignIn: ({email}) => {
+        const response = require('../E2E/apiMocks/beginSignin.json');
+        response.onyxData.forEach((data) => {
+            if (data.key !== 'credentials') {
+                return;
+            }
+            // eslint-disable-next-line no-param-reassign
+            data.value.login = email;
+        });
+        return response;
+    },
+    SigninUser: ({email}) => {
+        const response = require('../E2E/apiMocks/signinUser.json');
+        response.onyxData.forEach((data) => {
+            if (data.key !== 'session') {
+                return;
+            }
+            // eslint-disable-next-line no-param-reassign
+            data.value.email = email;
+        });
+        return response;
+    },
```

Comment:
I was talking with @AndrewGable about this, and I don't think it's very ideal that there is logic happening in these methods. They should really just be basic passthrough methods (like the ones below). I understand that the logic right now is so that the response contains the data passed to the parameters of the API method, so this is the change I would propose:

1. Change all of the `.json` files containing response data to `.js` files which export a single method that returns the JSON. 
2. Then for these two API commands, the `email` can be passed to the `.js` file so that the function can include `email` in the returned JSON

I hope that makes sense! I think it's OK for now, so I will go ahead and merge, but it could be good to clean that up in a second PR.